docs/source/dev-guide/deep-dive-solvers.md

```diff
@@ -0,0 +1,526 @@
+`specs=[MatchSpec("numpy")]`? Well, not that quick. The explicit instructions given by the user
+are only one part of the request we will send to the solver. Other pieces of implicit state are
+taken into account to build the final request. Namely, the state of your prefix. In total,
+these are the ingredients of the solver request:
+
+1. Packages already present in your environment, if you are not _creating_ a new one. This is
+   exposed through the `conda.core.prefix_data.PrefixData` class, which provides an iterator method
+   via `.iter_records()`. As we saw before, this yields `conda.models.records.PrefixRecord`
+   objects, a `PackageRecord` subclass for installed records.
+2. Past actions you have performed in that environment; the _History_. This is a journal of all the
+   `conda install|update|remove` commands you have run in the past. In other words, the _specs_
+   matched by those previous actions will receive extra protections in the solver.
+3. Packages included in the _aggressive updates_ list. These packages are always included in any
+   requests to make sure they stay up-to-date under all circumstances.
+4. Packages pinned to a specific version, either via `pinned_packages` in your `.condarc` or defined
+   in a `$PREFIX/conda-meta/pinned` file.
+5. In new environments, packages included in the `create_default_packages` list. These specs are
+   injected in each `conda create` command, so the solver will see them as explicitly requested
+   by the user.
+6. And, finally, the specs the user is asking for. Sometimes this is explicit (e.g.
+   `conda install numpy`) and sometimes a bit implicit (e.g. `conda update --all` is telling the
+   solver to add all installed packages to the update list).
+
+All of those sources of information produce a number a of `MatchSpec` objects, which are then
+combined and modified in very specific ways depending on the command line flags and their origin
+(e.g. specs coming from the pinned packages won't be modified, unless the user asks for it
+explicitly). This logic is intricate and will be covered in the next sections. A more technical
+description is also available in {ref}`solver_state_specification`.
+
+```{figure} /img/solver-deep-dive-1.png
+:name: solver-local-variables
+
+Local variables affect the solving process explicitly and implicitly. As seen in
+{ref}`in the conda install deep dive <deep_dive_install>`, the main actor is the
+`conda.core.solve.Solver` class. Before invoking the SAT solver, we can describe nine steps:
+
+1.  Instantiate the `Solver` class with the user-requested packages and the active environment
+    (target prefix)
+2.  Call the `solve_for_transaction()` method on the instance, which calls `solve_for_diff()`.
+3.  Call `solve_final_state()`, which takes some more arguments from the CLI.
+4.  Under some circumstances, we can return early (e.g. the packages are already installed)
```

Comment:
Period needed at the end of this sentence, to match the others.

Reply:
```suggestion
4.  Under some circumstances, we can return early (e.g. the packages are already installed).
```